homeassistant/components/http/ban.py

```diff
@@ -76,12 +98,38 @@ async def ban_middleware(
         _LOGGER.error("IP Ban middleware loaded but banned IPs not loaded")
         return await handler(request)
 
+    ip_address_ = ip_address(request.remote)  # type: ignore[arg-type]
     if ip_bans_lookup := ban_manager.ip_bans_lookup:
         # Verify if IP is not banned
-        ip_address_ = ip_address(request.remote)  # type: ignore[arg-type]
         if ip_address_ in ip_bans_lookup:
             raise HTTPForbidden
-
+    # Verify the whole subnet isn't banned
+    if banned_networks := ban_manager.banned_networks:
+        for banned_network in banned_networks:
+            if ip_address_ in banned_network:
+                if ban_manager.notify_bans or ban_manager.log_bans:
+                    hass = ban_manager.hass
+                    remote_host = request.remote
+                    with suppress(herror):
+                        remote_host, _, _ = await hass.async_add_executor_job(
+                            gethostbyaddr, request.remote
+                        )
+
+                    base_msg = f"Prevented access attempt from {remote_host} ({ip_address_}) which is in banned network {banned_network}"
+                    if ban_manager.notify_bans:
+                        # Circular import with websocket_api
+                        # pylint: disable=import-outside-toplevel
+                        from homeassistant.components import persistent_notification
```

Comment:
We generally don't add new persistent notifications

Reply:
OK happy to remove this if it's seen as an issue
I could make the default setting "no logging" - would that be OK?